### PR TITLE
Holiday Monsters Part I: Reindeer

### PR DESCRIPTION
**Pull request by Theawesomeboophis**
#### Summary
SUMMARY: Content "Adds Reindeer as well as Zombie and Fawn Variants"
#### Purpose of change
Adds Reindeer as part of #45672 
#### Describe the solution
Adds Reindeer as well as fawn and undead variants, all of which only spawn in winter and autumn.
#### Describe alternatives you've considered
Not Doing This.
#### Testing
Spawns, Looks right, all variants tested.
Note: Melee Skill bug fixed, removed melee skill.

### CODE REVIEW

**Comment by Theawesomeboophis** [comment]
DEBUG    : (json-error)
Json error: data/mods//dda/../../json/monsters/mammal.json:2403:19: assignment does not update value

    "symbol": "r",
    "morale": 42,
    "melee_skill":
                  ^
                   3,
    "melee_dice": 3,
    "melee_dice_sides": 4,


 FUNCTION : void report_strict_violation(const JsonObject&, const string&, const string&)
 FILE     : src/assign.h
 LINE     : 49
Error with melee skill.

File: data/json/monsters/mammal.json [hunk 1]
```diff
@@ -2351,5 +2351,62 @@
...
+    "copy-from": "mon_reindeer",
+    "name": { "str": "Reindeer fawn" },
+    "description": "A juvenile Caribou, it seems to be smaller than its adult counterparts, as well as having shorter horns.",
+    "volume": "30000 ml",
+    "weight": "120 g",
+    "hp": 35,
+    "speed": 93,
+    "symbol": "r",
+    "morale": 42,
+    "melee_skill": 3,
```
Comment (Fris0uman) [inline data/json/monsters/mammal.json:2403]:
```suggestion
```
You don't need it if you're already using copy-from and not changing the value

File: data/json/monsters/mammal.json [hunk 2]
```diff
@@ -2351,5 +2351,62 @@
...
+    "melee_dice": 3,
```
Comment (Fris0uman) [inline data/json/monsters/mammal.json:2404]:
```suggestion
```
same

File: data/json/monsters/mammal.json [hunk 3]
```diff
@@ -2351,5 +2351,62 @@
     "death_function": [ "NORMAL" ],
     "zombify_into": "mon_zolf",
     "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "KEENNOSE" ]
+  },
+  {
+    "id": "mon_reindeer",
+    "type": "MONSTER",
+    "name": { "str": "Reindeer" },
+    "description": "Also known as the Caribou, this deer is an icon of Christmas tradition.",
```
Comment (Fris0uman) [inline data/json/monsters/mammal.json:2359]: It's a bit weird to reference Christmas here. This is not a mmorpg with seasonal events.

File: data/json/monsters/mammal.json [hunk 4]
```diff
@@ -2351,5 +2351,62 @@
...
+    "baby_flags": [ "WINTER" ],
+    "special_attacks": [ [ "EAT_CROP", 60 ] ],
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM" ]
+  },
+  {
+    "id": "mon_reindeer_fawn",
+    "type": "MONSTER",
+    "copy-from": "mon_reindeer",
+    "name": { "str": "Reindeer fawn" },
+    "description": "A juvenile Caribou, it seems to be smaller than its adult counterparts, as well as having shorter horns.",
```
Comment (Fris0uman) [inline data/json/monsters/mammal.json:2396]:
```suggestion
    "description": "A juvenile Caribou, it seems to be smaller than its adult counterparts, as well as having shorter antlers.",
```

**Review by Fris0uman — COMMENTED** [review]
Do reindeer actually live in new england? And it seems that you forgot to make them spawn.

File: data/json/monsters/mammal.json [hunk 5]
```diff
@@ -2351,5 +2351,62 @@
...
+    "vision_night": 13,
+    "harvest": "mammal_large_leather",
+    "path_settings": { "max_dist": 14 },
+    "fear_triggers": [ "SOUND", "PLAYER_CLOSE" ],
+    "anger_triggers": [ "HURT", "MATING_SEASON" ],
+    "death_function": [ "NORMAL" ],
+    "zombify_into": "mon_zeindeer",
+    "reproduction": { "baby_monster": "mon_reindeer_fawn", "baby_count": 5, "baby_timer": 340 },
+    "biosignature": { "biosig_item": "feces_manure", "biosig_timer": 12 },
+    "baby_flags": [ "WINTER" ],
```
Comment (Fris0uman) [inline data/json/monsters/mammal.json:2387]: Do reindeer reproduce during winter? That's a bit weird

File: data/json/monsters/mammal.json (hunk 1 above)
Comment (Theawesomeboophis) [reply data/json/monsters/mammal.json:2403]: Reindeer do not live in New England, but they do live in the surrounding areas, and it is highly possible that Reindeer pass through New England during migration.

File: data/json/monsters/mammal.json (hunk 5 above)
Comment (Theawesomeboophis) [reply data/json/monsters/mammal.json:2387]: They do during autumn and  early winter.

**Comment by Positronic-Girl** [comment]
I'm pretty sure reindeer don't live in the new england area, its a bit too south for their range

**Comment by Theawesomeboophis** [comment]
https://en.wikipedia.org/wiki/Boreal_woodland_caribou
They used to, but were all killed off with none remaining in the area.

**Comment by Positronic-Girl** [comment]
> https://en.wikipedia.org/wiki/Boreal_woodland_caribou
> They used to, but were all killed off with none remaining in the area.
> According to some sources, they used to live in New Hampshire, which is part of New England.
> Note: They are listed as extirpated, which is why this was edited, there are no known currently living Caribou in the area.
> The killing of the Caribou in New England was due to wolf predation and occurred in the 1970's. However, although numbers were lowered in the 1970's, at least 9000 lived in New England in 2018, by April 2018, only 3 remained.

I feel like if they are down to only 3 remaining, and almost extinct in the area, that it wouldn't really be prime something to add here.

**Comment by Theawesomeboophis** [comment]
> > https://en.wikipedia.org/wiki/Boreal_woodland_caribou
> > They used to, but were all killed off with none remaining in the area.
> > According to some sources, they used to live in New Hampshire, which is part of New England.
> > Note: They are listed as extirpated, which is why this was edited, there are no known currently living Caribou in the area.
> > The killing of the Caribou in New England was due to wolf predation and occurred in the 1970's. However, although numbers were lowered in the 1970's, at least 9000 lived in New England in 2018, by April 2018, only 3 remained.
> 
> I feel like if they are down to only 3 remaining, and almost extinct in the area, that it wouldn't really be prime something to add here.

Scientist have stated that it is likely they will return in the future, also, due to conservation efforts, human-bred reindeer have been being slowly released back into the area since 2019.

**Comment by ZhilkinSerg** [comment]
What's that Santaclysm now?

**Comment by Theawesomeboophis** [comment]
Kinda, I'm adding some festive monsters, all others will be added to mods
though.

**Comment by Theawesomeboophis** [comment]
> What's that Santaclysm now?
Kinda, I'm adding some festive monsters, all others will be added to mods
though.

**Comment by Theawesomeboophis** [comment]
Reindeer are now mountable.

**Comment by Theawesomeboophis** [comment]
How do I fix the build matrix checks?

**Comment by anothersimulacrum** [comment]
Dno't worry about it, the failure is not your fault.

**Comment by Theawesomeboophis** [comment]
Ok, Thanks.

File: data/json/monsters/mammal.json [hunk 6]
```diff
@@ -2351,5 +2351,60 @@
...
+    "path_settings": { "max_dist": 14 },
+    "fear_triggers": [ "SOUND", "PLAYER_CLOSE" ],
+    "anger_triggers": [ "HURT", "MATING_SEASON" ],
+    "death_function": [ "NORMAL" ],
+    "zombify_into": "mon_zeindeer",
+    "reproduction": { "baby_monster": "mon_reindeer_fawn", "baby_count": 5, "baby_timer": 340 },
+    "biosignature": { "biosig_item": "feces_manure", "biosig_timer": 12 },
+    "baby_flags": [ "WINTER", "AUTUMN" ],
+    "special_attacks": [ [ "EAT_CROP", 60 ] ],
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "CATTLEFODDER", "PET_MOUNTABLE" ]
```
Comment (kevingranade) [inline data/json/monsters/mammal.json:2389]: Wild animals should not be tameable.
```suggestion
    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM" ]
```

**Pull request merged**